cpp/src/copying/concatenate.cu

```diff
@@ -337,6 +337,30 @@ std::unique_ptr<column> concatenate(std::vector<column_view> const& columns_to_c
   return experimental::type_dispatcher(type, concatenate_dispatch{columns_to_concat, mr, stream});
 }
 
+std::unique_ptr<experimental::table> concatenate(std::vector<table_view> const& tables_to_concat,
+                                                 rmm::mr::device_memory_resource* mr,
+                                                 cudaStream_t stream)
+{
+  if (tables_to_concat.size() == 0) { return std::make_unique<experimental::table>(); }
+
+  table_view const first_table = tables_to_concat.front();
+  CUDF_EXPECTS(std::all_of(tables_to_concat.begin(),
+                           tables_to_concat.end(),
+                           [&first_table](auto const& t) {
+                             return t.num_columns() == first_table.num_columns() &&
+                                    have_same_types(first_table, t);
+                           }),
+               "Mismatch in table columns to concatenate.");
+
+  std::vector<std::unique_ptr<column>> concat_columns;
+  for (size_type i = 0; i < first_table.num_columns(); ++i) {
+    std::vector<column_view> cols;
+    for (auto& t : tables_to_concat) { cols.emplace_back(t.column(i)); }
```

Comment:
Seems like this could be a `transform` with a `back_inserter`?
```cpp
std::transform(tables_to_concat.cbegin(),
               tables_to_concat.cend(),
               std::back_inserter(cols),
               [i](auto const& t) { return t.column(i); });
```

Reply:
Good suggestions. Didn't notice these since I just cut-pasted this chunk from the old function body and only modified it to add the `stream` parameter :sweat_smile: